File: eMovieFinder/eMovieFinder.Model/Entities/Director.cs
using eMovieFinder.Model.Utilities;
using System;

namespace eMovieFinder.Model.Entities
{
    public class Director : TimeStampObjectViewModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public string? FormattedDirectorRealName => $"{FirstName} {LastName}";
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/entities/director.dart
import 'package:json_annotation/json_annotation.dart';

import '../utilities/time_stamp_object.dart';

part 'director.g.dart';

@JsonSerializable()
class Director extends TimeStampObject {
  final int? id;
  final String? firstName;
  final String? lastName;
  final DateTime? birthDate;
  final String? formattedDirectorRealName;

  Director({
    super.creationDate,
    super.modifiedDate,
    super.createdById,
    super.modifiedById,
    super.formattedCreationDate,
    super.formattedModifiedDate,
    this.id,
    this.firstName,
    this.lastName,
    this.birthDate,
    this.formattedDirectorRealName
  });

  factory Director.fromJson(Map<String, dynamic> json) => _$DirectorFromJson(json);
  Map<String, dynamic> toJson() => _$DirectorToJson(this);
}